chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -794,14 +780,16 @@ async def generate_unsigned_spendbundle(
         new_owner: Optional[bytes] = None,
         new_did_inner_hash: Optional[bytes] = None,
         trade_prices_list: Optional[Program] = None,
+        nft_coin: NFTCoinInfo = None,
     ) -> Tuple[SpendBundle, Optional[TransactionRecord]]:
-        if coins is None or len(coins) > 1:
-            # Make sure the user is specifying which specific NFT coin to use
-            raise ValueError("NFT spends require a single selected coin")
-        elif len(payments) > 1:
-            raise ValueError("NFTs can only be sent to one party")
-        else:
-            nft_coin = [c for c in self.my_nft_coins if c.coin in coins][0]
+        if not nft_coin:
+            if coins is None or len(coins) > 1:
+                # Make sure the user is specifying which specific NFT coin to use
+                raise ValueError("NFT spends require a single selected coin")
+            elif len(payments) > 1:
+                raise ValueError("NFTs can only be sent to one party")
+            else:
+                nft_coin = [c for c in self.my_nft_coins if c.coin in coins][0]
```

Comment:
No need to create a list and index it.  Though this will give a different exception on failure.

```suggestion

            nft_coin = next(c for c in self.my_nft_coins if c.coin in coins)
```